src/BigT/Query.java

```diff
@@ -0,0 +1,43 @@
+package BigT;
+
+import diskmgr.PCounter;
+
+public class Query {
+    public Query(String bigtName, int type, int orderType, String rowFilter, String columnFilter, String valueFilter, int numBuf) {
+
+        String bigTable = bigtName+"_"+type;
+
+        try {
+
+            // Calling the constructor with the data
+
+            bigt table = new bigt(bigtName, type);
+
+            // Reading the data inserted
+
+            //buffer pages gets used up and we get buffer manager "BUFMGR: BUFFER_EXCEEDED" exception so use 3/4*numbuf
+            Stream stream = table.openStream(bigtName, orderType, rowFilter, columnFilter, valueFilter, (3*numBuf)/4);
```

Comment:
I think you should use it completely. We need to make sure the input is <= buff pool size.